packages/cinterop/src/darwin/kotlin/io/realm/internal/interop/RealmInterop.kt

```diff
@@ -1871,6 +1893,214 @@ actual object RealmInterop {
+            destinationState.nativeValue,
+            staticCFunction<COpaquePointer?, realm_flx_sync_subscription_set_state_e, Unit> { userData, state ->
+                val callback = safeUserData<SubscriptionSetCallback>(userData)
+                callback.onChange(CoreSubscriptionSetState.of(state))
+            },
+            StableRef.create(callback).asCPointer(),
+            staticCFunction { userdata ->
+                disposeUserData<(SubscriptionSetCallback) -> Unit>(userdata)
+            }
+        )
+    }
+
+    actual fun realm_sync_subscriptionset_version(subscriptionSet: RealmBaseSubscriptionSetPointer): Long {
+        return realm_wrapper.realm_sync_subscription_set_version(subscriptionSet.cptr())
+    }
+
+    actual fun realm_sync_subscriptionset_state(subscriptionSet: RealmBaseSubscriptionSetPointer): CoreSubscriptionSetState {
+        val value: realm_flx_sync_subscription_set_state_e =
+            realm_wrapper.realm_sync_subscription_set_state(subscriptionSet.cptr())
+        return CoreSubscriptionSetState.of(value)
+    }
+
+    actual fun realm_sync_subscriptionset_error_str(subscriptionSet: RealmBaseSubscriptionSetPointer): String? {
+        return realm_wrapper.realm_sync_subscription_set_error_str(subscriptionSet.cptr())?.toKString()
+    }
+
+    actual fun realm_sync_subscriptionset_size(subscriptionSet: RealmBaseSubscriptionSetPointer): Long {
+        return realm_wrapper.realm_sync_subscription_set_size(subscriptionSet.cptr()).toLong()
+    }
+
+    actual fun realm_sync_subscription_at(
+        subscriptionSet: RealmBaseSubscriptionSetPointer,
+        index: Long
+    ): RealmSubscriptionPointer {
+        return CPointerWrapper(realm_wrapper.realm_sync_subscription_at(subscriptionSet.cptr(), index.toULong()))
+    }
+
+    actual fun realm_sync_find_subscription_by_name(
+        subscriptionSet: RealmBaseSubscriptionSetPointer,
+        name: String
+    ): RealmSubscriptionPointer? {
+        val ptr = realm_wrapper.realm_sync_find_subscription_by_name(subscriptionSet.cptr(), name)
+        return nativePointerOrNull(ptr)
+    }
+
+    actual fun realm_sync_find_subscription_by_query(
+        subscriptionSet: RealmBaseSubscriptionSetPointer,
+        query: RealmQueryPointer
+    ): RealmSubscriptionPointer? {
+        val ptr = realm_wrapper.realm_sync_find_subscription_by_query(subscriptionSet.cptr(), query.cptr())
+        return nativePointerOrNull(ptr)
+    }
+
+    actual fun realm_sync_subscriptionset_refresh(subscriptionSet: RealmSubscriptionSetPointer): Boolean {
+        return realm_wrapper.realm_sync_subscription_set_refresh(subscriptionSet.cptr())
+    }
+
+    actual fun realm_sync_make_subscriptionset_mutable(
+        subscriptionSet: RealmSubscriptionSetPointer
+    ): RealmMutableSubscriptionSetPointer {
+        return CPointerWrapper(
+            realm_wrapper.realm_sync_make_subscription_set_mutable(subscriptionSet.cptr()),
+            managed = false
```

Comment:
Because the `MutableRealm` keeps the Realm file open. So if we don't manually release it, it won't be closed when you close the Realm, but when the GC kicks in. This can prevent deleting the entire Realm file.

Reply:
But then we don't close it if the whole realm is GC'ed. I guess this calls for solution just like we have with the realm-references where we do both. I assume that the likelyhood of people closing the realm is bigger than GC'ing the whole realm, so probably OK for now, but we should create an issue for the leak.